pymeasure/instruments/thyracont/smartline_v2.py

```diff
@@ -0,0 +1,525 @@
+                answer += got
+        got = answer.decode('ascii')
+        # Raises from time to time an UnicodeDecodeError or pyvisa.VisaIOError
+        if got[3] == "7":
+            raise ConnectionError(self.errors[got[8:-1]])
+        if command is not None and got[4:6] != command:
+            raise ConnectionError(f"Wrong response to {command}: '{got}'.")
+        if calculateChecksum(got[:-1]) != got[-1]:
+            raise ConnectionError("Response checksum is wrong.")
+        return got[8:-1]
+
+    def check_set_errors(self):
+        """Check the errors after setting a property."""
+        self.read()
+        return []  # no error happened
+
+    " Main commands"
+
+    range = Instrument.measurement(
+        "0MR00", """Get the measurement range in mbar.""",
+        preprocess_reply=lambda r: r[1:], separator="L")
+
+    pressure = Instrument.measurement(
+        "0MV00", """Get the current pressure of the default sensor in mbar""",
+        preprocess_reply=lambda r: r.replace("UR", "0").replace("OR", "inf"),
+    )
+
+    # def Relays
+
+    display_unit = Instrument.control(
+        get_command="0DU00",
+        set_command="2DU%s",
+        docs="""Control the unit shown in the display. ('mbar', 'Torr', 'hPa')""",
+        values=['mbar', 'Torr', 'hPa'],
+        validator=validators.strict_discrete_set,
+        set_process=compose_data,
+        check_set_errors=True,
+    )
+
+    display_orientation = Instrument.control(
+        "0DO00", "2DO01%i",
+        """Control the orientation of the display in relation to the pipe ('top', 'bottom').""",
+        values={"top": 0, "bottom": 1},
+        map_values=True,
+        validator=validators.strict_discrete_set,
+        check_set_errors=True,
+    )
+
+    display_data = Instrument.control(
+        "0DD00", "2DD01%i",
+        """Control the display data source (strict SOURCES).""",
+        values=source,
+        cast=int,
+        map_values=True,
+        validator=validators.strict_discrete_set,
+        check_set_errors=True,
+    )
+
+    def setHigh(self, high=""):
+        """Set the high pressure to `high` pressure in mbar."""
+        self.ask_manually(2, "AH", high)
+
+    def setLow(self, low=""):
+        """Set the low pressure to `low` pressure in mbar."""
+        self.ask_manually(2, "AL", low)
+
+    " Sensor parameters"
+
+    def get_sensor_transition(self):
+        """
+        Get the current sensor transition between sensors.
+
+        return interpretation:
+            - direct
+                switch at 1 mbar.
+            - continuos
+                switch between 5 and 15 mbar.
+            - F[float]T[float]
+                switch between low and high value.
+            - D[float]
+                switch at value.
+        """
+        got = self.ask_manually(0, "ST")
+        # VSR/VSL: 0 direct switch at 1 mbar, 1 continuios between 5 to 15 mbar
+        if got == "0":
+            return "direct"
+        elif got == "1":
```

Comment:
```suggestion
        elif got in ["1", "2"]:
```

Reply:
then you do not know, what the transition mode is.
I'll name it "continuous mode 2" or such.